### ISSUE TITLE
`valid-expect` false positive on using two matchers with one expect

### ISSUE THREAD

**Issue by golergka** [jest-community/eslint-plugin-jest#695]
I use expect to check that it throws an exception:

```Typescript
    const thrown = await expect(
        async () => await thing.doSomethingWrong()
    ).rejects

    thrown.toThrow(SomeKindOfError)
    thrown.toThrowErrorMatchingSnapshot()
```

I use the same `thrown` because I only want to trigger error once: calling the same method the second time will not necessarily trigger the same error because of inner logic, and I don't want to increase the amount of my unit tests twice for every time I want to check a specific error and it's details.

However, this code produces a `Expect must have a corresponding matcher call  jest/valid-expect` error, which is a false positive in this case.
